inference-engine/src/vpu/graph_transformer/src/middleend/passes/merge_permute_stages.cpp

```diff
@@ -55,12 +55,33 @@ class PassImpl : public Pass {
                                   permuteStage->output(0)->desc().dimsOrder());
     }
 
-    static bool isTrivialPermute(const PermutationIndexVector& permuteDims) {
-        for (size_t i = 0; i < permuteDims.size(); ++i)
-            if (i != permuteDims[i])
-                return false;
+    static bool isTrivialPermute(const PermutationIndexVector& permuteDims, const vpu::DimValues& dims) {
+        auto defDims = DimsOrder::fromNumDims(dims.size());
+        std::vector<int> dimsVector(dims.size());
+        for (const auto& dim : dims) {
+            dimsVector[defDims.dimInd(dim.first)] = dim.second;
+        }
+
+        size_t i;
+        bool swapAdjacentDims, dimIsOne = false;
+
+        for (i = 0; i < permuteDims.size(); ++i)
+            if (i != permuteDims[i]) {
+                swapAdjacentDims = permuteDims[i] == (i + 1) && permuteDims[i + 1] == i;
+                dimIsOne = dimsVector[i] == 1 || dimsVector[i + 1] == 1;
+                if (swapAdjacentDims && dimIsOne) {
+                        i++;
+                        continue;
+                } else {
+                    break;
```

Comment:
May be I'm not right, but you can replace `break` with `return false`, but ```if (i == permuteDims.size()){return true;}``` which follow bellow with `return true;`

Reply:
Yes, it’s more logical, thanks! 😄